### PR TITLE
[Merged by Bors] - Scene example: write file in a task

### LINKED ISSUE #5951: Revert change to scene saving example to demonstrate async API
This is a bad idea, and we should really not show usage of a blocking api (file writing) in a system.

_Originally posted by @mockersf in https://github.com/bevyengine/bevy/issues/5949#issuecomment-1243040910_

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Fix #5951 
- Improve on #5949 by not risking blocking a system

## Solution

- Wrap file writing in a task

### CODE REVIEW

**Comment by rparrett** [comment]
This might be slightly silly, but do you think it would be a good idea to mention in a comment that this doesn't work on the web?

**Review by devil-ira — APPROVED** [review]

**Comment by mockersf** [comment]
> This might be slightly silly, but do you think it would be a good idea to mention in a comment that this doesn't work on the web?

Good point! added a comment and disabled that part in wasm

File: examples/scene/scene.rs [hunk 1]
```diff
@@ -104,17 +104,23 @@ fn save_scene_system(world: &mut World) {
...
-                .serialize_ron(type_registry)
-                .and_then(|data| file.write(data.as_bytes()).map_err(|err| err.into()))
+    let serialized_scene = scene.serialize_ron(type_registry).unwrap();
+
+    // Showing the scene in the console
+    info!("{}", serialized_scene);
+
+    // Writing the scene to a new file. Using a task to avoid calling the filesystem APIs in a system
+    // as they are blocking
+    // This can't work in WASM as there is not filesystem access
```
Comment (devil-ira) [inline examples/scene/scene.rs:114]:
```suggestion
    // This can't work in WASM as there is no filesystem access
```

**Review by rparrett — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Thank you very much @mockersf; I agree that this is a better pattern to teach.

**Review by IceSentry — APPROVED** [review]

**Review by afonsolage — APPROVED** [review]

**Pull request closed without merging**